src/processor/operator/hash_join/join_hash_table.cpp

```diff
@@ -125,24 +125,29 @@ void JoinHashTable::buildHashSlots() {
     }
 }
 
-void JoinHashTable::probe(const std::vector<ValueVector*>& keyVectors, ValueVector* hashVector,
-    ValueVector* tmpHashVector, uint8_t** probedTuples) {
+void JoinHashTable::probe(const std::vector<ValueVector*>& keyVectors, ValueVector& hashVector,
+    ValueVector& tmpHashResultVector, uint8_t** probedTuples) {
     KU_ASSERT(keyVectors.size() == keyTypes.size());
     if (getNumTuples() == 0) {
         return;
     }
     if (!discardNullFromKeys(keyVectors)) {
         return;
     }
-    function::VectorHashFunction::computeHash(keyVectors[0], hashVector);
+    SelectionVector selVec;
```

Comment:
I don't recommend creating a new selVector per probe. I would generate one in local state at compile time.